python/tests/test_parameter.py

```diff
@@ -23,3 +23,20 @@ def test_linear(state):
 
     linear = flyft.parameter.LinearParameter(3.0, 1.0, 4.5)
     assert linear(state) == pytest.approx(3.0+4.5*(2.0-1.0))
+
+def test_squareroot(state):
+    state.time = 2.0
+    squareroot = flyft.parameter.SquareRootParameter(3.0, 1.0, 4.5)
+    assert squareroot(state) == pytest.approx(np.sqrt(3.0*3.0+4.5*(2.0-1.0)))
+
+def test_squareroot_exception(state):
+    state.time = -3.0
+    squareroot = flyft.parameter.SquareRootParameter(1.0, 1.0, 1.0)
+    with pytest.raises(ValueError):
+        squareroot(state)
+    state.time = 2.0
+    squareroot = flyft.parameter.SquareRootParameter(1.0, 1.0, -3.0)
```

Comment:
One small thing here to make it simpler (& also test the setter):
```suggestion

    state.time = 2.0
    squareroot.rate = -3.0
```